src/main/java/org/broadinstitute/hellbender/tools/PrintReadsHeader.java

```diff
@@ -33,11 +32,9 @@ public boolean requiresReads() {
     public void traverse() {
         final SAMFileHeader bamHeader = getHeaderForReads();
 
-        try ( final PrintWriter outputWriter = new PrintWriter(outputFile) ) {
+        try ( final PrintWriter outputWriter = new PrintWriter(outputFile.getOutputStream()) ) {
             final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
             codec.encode(outputWriter, bamHeader);
-        } catch (FileNotFoundException e ) {
-            throw new UserException.CouldNotCreateOutputFile("Error writing reads header to " + outputFile, e);
```

Comment:
Same issue here -- need to check the error state of the `PrintWriter`. If exceptions are being suppressed internally in the writer then you're supposed to call `checkError()`, I believe.

Reply:
Right - in this case I think we should always have been calling checkError, independent of the change in this PR, since we were always using `PrintWriter` (the GATKPath getOutputStream does handle IOException). So I think the right thing to do here is to add the checkError call.

Reply:
Also switching this to `OutputStreamWriter`.